File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/ui/guides/ImplementingOwnCapabilities.java
package io.getstream.chat.docs.java.ui.guides;

import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

import io.getstream.chat.android.models.ChannelCapabilities;
import io.getstream.chat.android.ui.common.state.messages.MessageMode;
import io.getstream.chat.android.ui.common.state.messages.composer.MessageComposerState;
import io.getstream.chat.android.ui.feature.messages.composer.MessageComposerView;

/**
 * [Implementing Own Capabilities](https://getstream.io/chat/docs/sdk/android/ui/guides/implementing-own-capabilities/)
 */
public class ImplementingOwnCapabilities {

    private MessageComposerView messageComposerView;

    public void customCapabilities() {
        Set<String> customOwnCapabilities = new HashSet<>();
        customOwnCapabilities.add(ChannelCapabilities.SEND_MESSAGE);
        customOwnCapabilities.add(ChannelCapabilities.SEND_LINKS);
        customOwnCapabilities.add(ChannelCapabilities.UPLOAD_FILE);
        customOwnCapabilities.add(ChannelCapabilities.SEND_TYPING_EVENTS);
        customOwnCapabilities.add(ChannelCapabilities.TYPING_EVENTS);

        MessageComposerState messageComposerState = new MessageComposerState(
                "",
                Collections.emptyList(),
                null,
                Collections.emptyList(),
                Collections.emptyList(),
                Collections.emptyList(),
                Collections.emptyList(),
                0,
                MessageMode.Normal.INSTANCE,
                false,
                customOwnCapabilities,
                false
        );

        messageComposerView.renderState(messageComposerState);
    }
}

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/ui/guides/ImplementingOwnCapabilities.kt
package io.getstream.chat.docs.kotlin.ui.guides

import io.getstream.chat.android.models.ChannelCapabilities
import io.getstream.chat.android.ui.common.state.messages.composer.MessageComposerState
import io.getstream.chat.android.ui.feature.messages.composer.MessageComposerView

/**
 * [Implementing Own Capabilities](https://getstream.io/chat/docs/sdk/android/ui/guides/implementing-own-capabilities/)
 */
class ImplementingOwnCapabilities {

    private lateinit var messageComposerView: MessageComposerView

    fun customCapabilities() {
        val customOwnCapabilities = setOf(
            ChannelCapabilities.SEND_MESSAGE,
            ChannelCapabilities.SEND_LINKS,
            ChannelCapabilities.UPLOAD_FILE,
            ChannelCapabilities.SEND_TYPING_EVENTS
        )

        val messageComposerState = MessageComposerState(
            ownCapabilities = customOwnCapabilities
        )

        messageComposerView.renderState(messageComposerState)
    }
}
